pandas/core/nanops.py

```diff
@@ -812,15 +812,21 @@ def unique1d(values):
     return uniques
 
 
-def _checked_add_with_arr(arr, b):
+def _checked_add_with_arr(arr, b, arr_nans=None, b_nans=None):
```

Comment:
the problem is future readers will just be plain confused

change this to compute the mask internally like all other functions